src/NuGet.Core/NuGet.Protocol/UserAgentStringBuilder.cs

```diff
@@ -90,26 +86,35 @@ public string Build()
                     UserAgentWithOSDescriptionAndVisualStudioSKUTemplate,
                     _clientName,
                     NuGetClientVersion, /* NuGet version */
-                    osDescription, /* OS version */
+                    _osInfo, /* OS version */
                     _vsInfo);  /* VS SKU + version */
             }
         }
 
-        private string GetOSVersion()
+        internal static string GetOS()
         {
-            if (_osInfo == null)
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return OSPlatform.Windows.ToString();
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                return OSPlatform.Linux.ToString();
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return OSPlatform.OSX.ToString();
+            }
+#if NETCOREAPP
+            else if (OperatingSystem.IsBrowser())
             {
-#if !IS_CORECLR
-                // When not on CoreClr and no OSDescription was provided,
-                // we will set it ourselves.
-                _osInfo = Environment.OSVersion.ToString();
-#else
-                // When on CoreClr, one should use the .WithOSDescription() method to set it.
-                _osInfo = string.Empty;
+                return "BROWSER";
+            }
 #endif
+            else
```

Comment:
It looks like [OSPlatform supports `FreeBSD`](https://learn.microsoft.com/dotnet/api/system.runtime.interopservices.osplatform?view=net-8.0) OS also, should we add support for that OS?